src/prefect/engine/task_runner.py

```diff
@@ -582,20 +582,20 @@ def get_task_inputs(
             # construct task inputs
             if edge.key is not None:
                 handlers[edge.key] = handler = getattr(
-                    edge.upstream_task, "result_handler", None
+                    edge.upstream_task, "result_handler", None # todo: here
                 )
                 task_inputs[  # type: ignore
                     edge.key
-                ] = upstream_state._result.to_result(  # type: ignore
+                ] = upstream_state._result.to_result(  # type: ignore # todo: here
```

Comment:
This step does _two_ things:

- reads the underlying result
- ensures that we use the user's result handler when performing the read (this is the "hydration" step)